tests/ballerina-test/src/test/java/org/ballerinalang/test/nativeimpl/functions/io/socket/MockSecureSocketServer.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.test.nativeimpl.functions.io.socket;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.URL;
+import java.nio.file.Paths;
+import javax.net.ssl.SSLServerSocketFactory;
+
+/**
+ * This server socket will use to mock the secure backend.
+ */
+public class MockSecureSocketServer {
```

Comment:
Can we reuse the MockSocketServer impl here ? so that secure server will be an extension of it ?

Reply:
Internal implementation is different in both cases.